packages/e2e-tests/specs/editor/various/change-detection.test.js

```diff
@@ -348,4 +327,23 @@ describe( 'Change detection', () => {
 		// Verify that the post is not dirty.
 		await assertIsDirty( false );
 	} );
+
+	it( 'should not prompt to confirm unsaved changes when trashing an existing post', async () => {
+		// Enter title.
+		await page.type( '.editor-post-title__input', 'Hello World' );
+
+		// Save
+		await saveDraft();
+
+		// Trash post.
+		await openDocumentSettingsSidebar();
+		await page.click( '.editor-post-trash.components-button' );
+
+		// Wait for "Saved" to confirm save complete.
+		await page.waitForSelector( '.editor-post-saved-state.is-saved' );
+
+		// Make sure redirection happens.
+		await page.waitForNavigation();
```

Comment:
I'm not sure if it's going to be an issue, but I recall there being some potential for race conditions in Puppeteer when it's expected that an action can be a direct result of another, i.e. around the timing of when the previous promise is resolved and the next one "bound" (likely related to the fact that there can be a delay in [communicating](https://chromedevtools.github.io/devtools-protocol/) between the page and the parent).

https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageclickselector-options

>Bear in mind that if click() triggers a navigation event and there's a separate page.waitForNavigation() promise to be resolved, you may end up with a race condition that yields unexpected results. The correct pattern for click and wait for navigation is the following:

I'm not sure if that's something we can expect to happen here, but worth considering.

Reply:
You're right, I've made it more resilient.